guppylang/std/builtins.py

```diff
@@ -565,7 +572,9 @@ def __trunc__(self: float) -> float: ...
 
 
 @guppy.extend_type(list_type_def)
-class List:
+class list(Generic[_T]):
+    """Mutable sequence items with homogeneous types."""
```

Comment:
Just checking that a `list` can't have `hetrogeneous` types? Is it just the fact that the length can change at compile time that distinguishes a list from a mutable `array`?

Reply:
>  Is it just the fact that the length can change at runtime that distinguishes a list from a mutable array?

Yes, that's the only difference